ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotBlockRetriever.java

```diff
@@ -147,9 +126,10 @@ private void handleContestedPivotBlock(final long contestedBlockNumber) {
         contestedBlockNumber, contestedBlockNumber - pivotBlockNumberResetDelta)) {
       LOG.info("Received conflicting pivot blocks for {}.", contestedBlockNumber);
 
-      final int retryCount = confirmationTasks.size();
-      if (retryCount > maxPivotBlockResets
-          || pivotBlockNumber.get() <= BlockHeader.GENESIS_BLOCK_NUMBER) {
+      if (confirmationTasks.size() > SUSPICIOUS_NUMBER_OF_PIVOT_BLOCK_RESETS) {
+        LOG.warn("Several attempts have been made without finding a pivot block");
```

Comment:
I'd print a warning periodically (every `SUSPICIOUS_NUMBER_OF_PIVOT_BLOCK_RESETS` retries?) rather than on every individual retry.  Also, `confirmationTasks` can now grow very large, so we'll probably need to rework how we track tasks so we don't hold onto every confirmation task we created.  

Alternatively, I wonder if we could alleviate the problem of prematurely switching to full sync by just increasing `DEFAULT_MAX_PIVOT_BLOCK_RESETS` and making sure we retry fast sync here: https://github.com/hyperledger/besu/blob/7fe1d4796e52fa1ce6099f402cc05f1165607335/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DefaultSynchronizer.java#L166 ...